src/java/org/apache/cassandra/auth/IInternodeAuthenticator.java

```diff
@@ -35,10 +36,51 @@
      */
     boolean authenticate(InetAddress remoteAddress, int remotePort);
 
+    /**
+     * Decides whether a peer is allowed to connect to this node.
+     * If this method returns false, the socket will be immediately closed.
+     *
+     * @param certificates client certificates
+     * @return true if the connection should be accepted, false otherwise.
+     */
+    default boolean authenticate(Certificate[] certificates)
+    {
+        return false;
+    }
+
+
+    /**
+     * Decides whether a peer is allowed to connect to this node.
+     * If this method returns false, the socket will be immediately closed.
+     *
+     * @param remoteAddress ip address of the connecting node.
+     * @param remorePort port of the connecting node.
+     * @param certificates client certificates
+     * @return true if the connection should be accepted, false otherwise.
+     */
+    default boolean authenticate(InetAddress remoteAddress, int remorePort, Certificate[] certificates)
+    {
+        if (certificates != null)
+        {
+            return authenticate(certificates);
+        }
+        return authenticate(remoteAddress, remorePort);
+    }
+
     /**
      * Validates configuration of IInternodeAuthenticator implementation (if configurable).
      *
      * @throws ConfigurationException when there is a configuration error.
      */
     void validateConfiguration() throws ConfigurationException;
+
+    /**
+     * Setup is called once upon system startup to initialize the IAuthenticator.
+     *
+     * For example, use this method to create any required keyspaces/column families.
+     */
+    default void setup()
```

Comment:
If somebody wanted to centralize all their client `IAuthenticator` and internode auth `IInternodeAuthenticator` by implementing both in the same class the compiler would forbid with an `inherits unrelated defaults for setup()` error. WDYT about renaming this `setupInternode` to make that possible.